mvebu: update 309-linksys-status-led.patch

ricsc opened the pull request:
Fix power led status, by using correct naming scheme.

Kernel 6.6 requires name to be in form led-<<name>>

robimarko commented:
Does this fix an actual error, or just compliance with the schema?

ricsc commented:
Yes it fixes power led not working due to led gpio probe failing.

robimarko commented:
Please sort out the formal issue that CI is complaining

ricsc commented:
Signed off should be correct now to pass CI

robimarko commented:
Please squash the commits, there should be no merge commit

anomeome commented:
Appears to not work on a test with a rango for me.

ricsc commented:
> Appears to not work on a test with a rango for me.

How did you test? Did you look at build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/linux-6.6.67/arch/arm/boot/dts/marvell/armada-385-linksys.dtsi to check if the patch was applied? 

Did you run make target/linux/clean? I tested on a 1200AC but since rango also includes armada-385-linksys.dtsi it should also work for rango 

Note: In upstream kernel the naming is correct, the openwrt patch however adds aliases along with the now incorrect name. This is because it was taken over from older kernel where it was still valid.

anomeome commented:
From a:

make target/linux/{clean,compile}

a dump of file:

[build_dir/target-arm_cortex-a9+neon-vfpv3_musl_eabi/linux-mvebu_cortexa9/linux-6.6.67/arch/arm/boot/dts/marvell/armada-385-linksys.dtsi](https://termbin.com/zpnh)

ricsc commented:
> From a:
> 
> make target/linux/{clean,compile}
> 
> a dump of file:
> 
> [build_dir/target-arm_cortex-a9+neon-vfpv3_musl_eabi/linux-mvebu_cortexa9/linux-6.6.67/arch/arm/boot/dts/marvell/armada-385-linksys.dtsi](https://termbin.com/zpnh)

Are you sure there is no physical problem with your led. Without the patch you will see an error message in kernel log.
Try this on your router: dmesg | grep gpio
Does it display and error telling you that leds-gpio probe failed? If no such error then it is working and you should find rango:white:power in /sysfs under leds. If this is all okay, then either you have some trigger setting etc. that turn is off on request or it is broken.

flygarn12 reviewed (approved):
Tested this on wrt3200acm/rango and power led now works as it did before. Even tried a reset and now the blinking also works again (had it set to static on after this bug arrived but then reset blinking don't work).

```
root@OpenWrt:~# dmesg | grep gpio
[    0.031751] gpio gpiochip0: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.031825] debugfs: Directory 'f1018100.gpio' with parent 'regmap' already present!
[    0.032047] gpio gpiochip1: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.032129] debugfs: Directory 'f1018140.gpio' with parent 'regmap' already present!
[    4.966234] gpio_button_hotplug: loading out-of-tree module taints kernel.
root@OpenWrt:~#
```
But on this device the kernel messages with gpio is the same with or without this patch.

Merry Christmas!

flygarn12 commented:
Once we have this in main it should probably be backported to 24.10 also.

flygarn12 commented:
> Signed off should be correct now to pass CI

You need to find out what your real name is...

robimarko commented:
I have reworded the commit title and description, but the commit is still missing your Signed-off-by, please add it.

ricsc committed:
mvebu: fix Linksys power LED

Kernel 6.6 requires LED node names to be prefixed via "led-", otherwise
probing the LED will fail, so update our downstream patch adding the LED.

Signed-off-by: Richard Schneidt <ricsc@users.noreply.github.com>
Link: https://github.com/openwrt/openwrt/pull/17330
Signed-off-by: Robert Marko <robimarko@gmail.com>

ricsc commented:
> I have reworded the commit title and description, but the commit is still missing your Signed-off-by, please add it.

OK, signed off via commandline instead of via git webif in comment. Hopefully no conflicts anymore.

robimarko commented:
Thanks! Rebased on top of main and merged!

neheb commented:
Only real reference I found to this being a problem is: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.2.1&id=f4237d74e8ed507ccd39cdd35458c30b8d5c577b

Is it possible to run dtshema in OpenWrt?

robimarko commented:
@neheb I think you are right, only dtschema enforces this.

I tried to reproduce the reported issue that this fixes but I was not able to using labels, etc but I was unable to.

That being said, this breaks nothing.
